File: backend/src/app/models/item.py
from sqlalchemy import Column, Integer, String, Text, DateTime, func, ForeignKey
from sqlalchemy.orm import relationship
from ..db.session import Base

class Item(Base):
    __tablename__ = "items"

    id = Column(Integer, primary_key=True, index=True)
    world_id = Column(Integer, ForeignKey("worlds.id", ondelete="CASCADE"), nullable=False, index=True)
    character_id = Column(Integer, ForeignKey("characters.id", ondelete="SET NULL"), index=True, nullable=True)
    location_id = Column(Integer, ForeignKey("locations.id", ondelete="SET NULL"), index=True, nullable=True)
    name = Column(String, nullable=False)
    description = Column(Text)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    world = relationship("World", back_populates="items")
    character = relationship("Character", back_populates="items")
    location = relationship("Location", back_populates="items")
    tags = relationship("ItemTag", back_populates="item", cascade="all, delete-orphan")

    # Relationships will be added later
    # character = relationship("Character", back_populates="items")
    # location = relationship("Location", back_populates="items")
    # tags = relationship("ItemTag", back_populates="item") 

File: frontend/src/types/item.ts
// Based on backend/src/app/schemas/item.py

import type { ItemTag } from './itemTag'; // Import ItemTag

export interface ItemBase {
  name: string;
  description?: string | null;
  character_id?: number | null;
  location_id?: number | null;
}

export interface ItemCreate extends ItemBase {
  world_id: number;
}

export interface ItemUpdate {
  name?: string;
  description?: string | null;
  character_id?: number | null;
  location_id?: number | null;
}

/**
 * Represents an item within a world.
 */
export interface Item extends ItemBase {
  id: number;
  world_id: number;
  created_at: string; // Assuming ISO date strings from API
  updated_at: string;
  tags?: ItemTag[]; // Optional list of tags associated with the item
  assigned_character_name?: string; // Added character name from backend
} 